evm/src/main/java/org/hyperledger/besu/evm/fluent/EVMExecutor.java

```diff
@@ -302,7 +302,10 @@ public EVMExecutor code(final Code code) {
   }
 
   public EVMExecutor code(final Bytes codeBytes, final Hash hash) {
-    this.code = new Code(codeBytes, hash);
+    if (evm == null) {
```

Comment:
Could we add a constructor for the Executor and make the EVM a final parameter on it? Seems like that is a safe thing to make a hard dependency for this class.